### PR TITLE
fix(conform-nvim): fix formatters_by_ft definitions

### PR DESCRIPTION
**Pull request by lucazz**
## 📑 Description

As requested by @Uzaaft on #1127 #1128, I'm merging all changes in a single PR.

This PR updates stevearc/conform.nvim's setup, addressing the deprecation warning that comes up when opening any yaml file while using the yaml pack:
```
deprecated[conform]: The nested {} syntax to run the first formatter has been replaced by the stop_after_first option (see :help conform.format).
Support for the old syntax will be dropped on 2025-01-01.
```

### CODE REVIEW

**Comment by Uzaaft** [comment]
@lucazz I’m about to head to bed so I cant deep dive into it, but based upon the warning you have this seem wrong.

**Review by mehalter — CHANGES_REQUESTED** [review]
This is incorrect, all of them should be 

```lua
{ "prettierd", "prettier", stop_after_first = true }
```

**Comment by Uzaaft** [comment]
@lucazz  Why did you add it to the places where it's not needed?

**Comment by Uzaaft** [comment]
Danke Danke

File: lua/astrocommunity/pack/go/init.lua [hunk 1]
```diff
@@ -136,7 +136,7 @@ return {
     optional = true,
     opts = {
       formatters_by_ft = {
-        go = { "goimports", "gofumpt" },
+        go = { "goimports", "gofumpt", stop_after_first = true },
```
Comment (mehalter) [inline lua/astrocommunity/pack/go/init.lua:139]:
```suggestion
        go = { "goimports", "gofumpt" },
```
This is incorrect

File: lua/astrocommunity/pack/markdown/init.lua [hunk 2]
```diff
@@ -36,8 +36,8 @@ return {
     optional = true,
     opts = {
       formatters_by_ft = {
-        markdown = { { "prettierd", "prettier" } },
-        ["markdown.mdx"] = { { "prettierd", "prettier" } },
+        markdown = { "prettierd", "prettier" },
```
Comment (mehalter) [inline lua/astrocommunity/pack/markdown/init.lua:39]:
```suggestion
        markdown = { "prettierd", "prettier", stop_after_first = true },
```

This needs it

File: lua/astrocommunity/pack/python-ruff/init.lua [hunk 3]
```diff
@@ -41,7 +41,7 @@ return {
     optional = true,
     opts = {
       formatters_by_ft = {
-        python = { "ruff_organize_imports", "ruff_format" },
+        python = { "ruff_organize_imports", "ruff_format", stop_after_first = true },
```
Comment (mehalter) [inline lua/astrocommunity/pack/python-ruff/init.lua:44]:
```suggestion
        python = { "ruff_organize_imports", "ruff_format" },
```

This is incorrect

File: lua/astrocommunity/pack/python/init.lua [hunk 4]
```diff
@@ -131,7 +131,7 @@ return {
     optional = true,
     opts = {
       formatters_by_ft = {
-        python = { "isort", "black" },
+        python = { "isort", "black", stop_after_first = true },
```
Comment (mehalter) [inline lua/astrocommunity/pack/python/init.lua:134]:
```suggestion
        python = { "isort", "black" },
```

Not this one

**Review by mehalter — CHANGES_REQUESTED** [review]

**Review by mehalter — APPROVED** [review]
Ok, I think this looks correct now, thanks! Just to clarify why the previous ones were incorrect. The nested structure indicated when to only stop at the first. The ones that were not previously nested indicated _multiple_ formatters so we very much do not want them to stop at the first

**Pull request merged**